src/identity.cc

```diff
@@ -826,36 +1046,84 @@ TRITONBACKEND_ModelInstanceExecute(
                 &input_memory_type_id));
-        if ((responses[r] == nullptr) ||
-            (input_memory_type == TRITONSERVER_MEMORY_GPU)) {
+#ifndef TRITON_ENABLE_GPU
+        if (input_memory_type == TRITONSERVER_MEMORY_GPU) {
           GUARDED_RESPOND_IF_ERROR(
               responses, r,
               TRITONSERVER_ErrorNew(
                   TRITONSERVER_ERROR_UNSUPPORTED,
                   "failed to get input buffer in CPU memory"));
+          LOG_MESSAGE(
+              TRITONSERVER_LOG_ERROR,
+              (std::string("request ") + std::to_string(r) +
+               ": failed to get input buffer in CPU memory, error "
+               "response sent")
+                  .c_str());
+          continue;
+        }
+#endif  // TRITON_ENABLE_GPU
+        if (responses[r] == nullptr) {
+          GUARDED_RESPOND_IF_ERROR(
+              responses, r,
+              TRITONSERVER_ErrorNew(
+                  TRITONSERVER_ERROR_UNSUPPORTED,
+                  "failed to get input buffer"));
+          LOG_MESSAGE(
+              TRITONSERVER_LOG_ERROR,
+              (std::string("request ") + std::to_string(r) +
+               ": failed to get input buffer, error "
+               "response sent")
+                  .c_str());
+          continue;
         }
 
+#ifdef TRITON_ENABLE_GPU
+        bool cuda_used = false;
+        GUARDED_RESPOND_IF_ERROR(
+            responses, r,
+            CopyBuffer(
+                input_name, input_memory_type, input_memory_type_id,
+                output_memory_type, output_memory_type_id, buffer_byte_size,
+                input_buffer,
+                reinterpret_cast<char*>(output_buffer) + output_buffer_offset,
+                instance_state->CudaStream(), &cuda_used));
+        cuda_copy |= cuda_used;
+
+        if (responses[r] == nullptr) {
+          GUARDED_RESPOND_IF_ERROR(
+              responses, r,
+              TRITONSERVER_ErrorNew(
+                  TRITONSERVER_ERROR_UNSUPPORTED, "failed to get copy buffer"));
+          LOG_MESSAGE(
+              TRITONSERVER_LOG_ERROR,
+              (std::string("request ") + std::to_string(r) +
+               ": failed to get copy buffer, error response "
+               "sent")
+                  .c_str());
+          continue;
+        }
+#else
         memcpy(
             reinterpret_cast<char*>(output_buffer) + output_buffer_offset,
             input_buffer, buffer_byte_size);
+#endif  // TRITON_ENABLE_GPU
         output_buffer_offset += buffer_byte_size;
       }
+    }
 
-      if (responses[r] == nullptr) {
-        LOG_MESSAGE(
-            TRITONSERVER_LOG_ERROR,
-            (std::string("request ") + std::to_string(r) +
-             ": failed to get input buffer in CPU memory, error response "
-             "sent")
-                .c_str());
-        continue;
-      }
+    uint64_t compute_end_ns = 0;
+    SET_TIMESTAMP(compute_end_ns);
+    max_compute_end_ns = std::max(max_compute_end_ns, compute_end_ns);
```

Comment:
You don't need to compare for the max timestamp because the `compute_end_ns` will always be more recent that `max_compute_end_ns`